### PR TITLE
bpo-46564: do not create frame object for super object

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue46564

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
Marking `DO-NOT-MERGE` as it's already part of https://github.com/python/cpython/pull/30992. There's more discussion about this on the issue tracker.

File: Objects/typeobject.c [hunk 1]
```diff
@@ -9096,17 +9096,14 @@ super_init(PyObject *self, PyObject *args, PyObject *kwds)
...
                             "super(): no current frame");
             return -1;
         }
-
-        PyCodeObject *code = PyFrame_GetCode(frame);
-        int res = super_init_without_args(frame, code, &type, &obj);
-        Py_DECREF(frame);
-        Py_DECREF(code);
+        PyCodeObject *code = cframe->f_code;
+        int res = super_init_without_args(cframe, code, &type, &obj);
```
Comment (Fidget-Spinner) [inline Objects/typeobject.c]:
```suggestion
        int res = super_init_without_args(cframe, cframe->f_code, &type, &obj);
```

**Review by Fidget-Spinner — COMMENTED** [review]
Please write a short news entry. I'll merge after that.

File: Objects/typeobject.c (hunk 1 above)
Comment (kumaraditya303) [reply Objects/typeobject.c]: Fixed

File: Misc/NEWS.d/next/Core and Builtins/2022-02-01-10-23-21.bpo-46564.6Xc2_H.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Do not create frame objects for creating :class:`super` object.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Core and Builtins/2022-02-01-10-23-21.bpo-46564.6Xc2_H.rst:1]:
```suggestion
Do not create frame objects for creating :class:`super` object. Patch by Kumar Aditya.
```

File: Misc/NEWS.d/next/Core and Builtins/2022-02-01-10-23-21.bpo-46564.6Xc2_H.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Do not create frame objects for creating :class:`super` object. Patch by Kumar Aditya.
```
Comment (Fidget-Spinner) [inline Misc/NEWS.d/next/Core and Builtins/2022-02-01-10-23-21.bpo-46564.6Xc2_H.rst]:
```suggestion
Do not create frame objects when creating :class:`super` object. Patch by Kumar Aditya.
```

**Review by Fidget-Spinner — APPROVED** [review]

**Comment by Fidget-Spinner** [comment]
Thanks!

**Pull request merged**